draft-ietf-quic-multipath.md

```diff
@@ -1353,15 +1357,21 @@ Loss or reordering can cause an endpoint to receive a MAX_PATH_ID frame with
 a smaller Maximum Path Identifier value than was previously received.
 MAX_PATH_ID frames that do not increase the path limit MUST be ignored.
 
-## PATHS_BLOCKED frames {#paths-blocked-frame}
+## PATHS_BLOCKED and PATH_CIDS_BLOCKED frames {#paths-and-cids-blocked-frame}
 
 A sender SHOULD send a PATHS_BLOCKED frame (type=0x15228c0d) when
 it wishes to open a path but is unable to do so due to the maximum path identifier
 limit set by its peer.
-Note that PATHS_BLOCKED frame is informational. Sending a PATHS_BLOCKED frame does not
-imply a particular action from the peer like updating the new Max Path ID value,
-but informs the peer that the maximum path identifier limit prevented the creation of new paths.
 
+A sender SHOULD send a PATH_CIDS_BLOCKED frame (type=0x15228c0e) when
```

Comment:
I think we should actually change this "SHOULD" to "can". There is no requirement in interoperability that require the sender to send this. Or if we want normative language we should use "MAY".

The same comment of course applies to the PATHS_BLOCKED frame of course. Sorry I overlooked this earlier.

Reply:
I'd like to vote for "SHOULD" here. Basically it's a recommendation action in this kind of situation.

BTW, PATHS_BLOCKED and PATH_CIDS_BLOCKED are very similar to STREAMS_BLOCKED and we could find a "SHOULD" in [RFC 9000 Section 19.14](https://datatracker.ietf.org/doc/html/rfc9000#name-streams_blocked-frames):

> A sender SHOULD send a STREAMS_BLOCKED frame (type=0x16 or 0x17) when it wishes to open a stream but is unable to do so due to the maximum stream limit set by its peer...

Reply:
Actually I think this is slightly different. In case of streams there is no really good reason to limit streams too much. Of course you want to have a limit to avoid excessive resource usage because that would open an attack, however, under normal operation you should not be blocked. In case of the number of paths, there can be other reasons to limit, e.g. you may have a policy to not use more than a small number of paths. Therefore I argue that MAY would make more sense here than SHOULD. However, I don't have a really strong opinion. I would like to hear other people's thoughts.

Reply:
That's fair enough. For the reason of anti-attack, I agree to use "can" here.